src/plugins/query_enhancements/server/connections/resource_manager_service.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BaseConnectionManager } from './managers/base_connection_manager';
+
+class ResourceManagerService {
+  private readonly managers: { [key: string]: BaseConnectionManager };
+  constructor() {
+    this.managers = {};
+  }
+
+  register(dataConnectionType: string, manager: BaseConnectionManager) {
+    if (this.managers[dataConnectionType] !== undefined) {
+      throw new Error(
+        `Manager for dataConnectionType ${dataConnectionType} is already registered. Unable to register another manager.`
+      );
+    }
+    this.managers[dataConnectionType] = manager;
+  }
+
+  getManager(dataConnectionType: string): BaseConnectionManager | undefined {
+    return this.managers[dataConnectionType];
+  }
+}
```

Comment:
is there a use case where we want to delete a manager? If so should we expose that as a method?

Reply:
i don't think that would be needed, registering manager happens one time during plugin setup, i don't see a reasonable trigger on when would a plugin want to delete a manager